alphagov/paas-bootstrap
# [#154517370] Fix GitHub OAuth config, and add Hector

## What

Currently the config for the main team is being lost every time atc
restarts. This is because we're not supposed to use `fly set-team` to
configure the "main" team (as discovered in [this github issue](concourse/concourse#1963)[1]. This
is also [covered in the docs](http://concourse.ci/teams.html#main-team).

This therefore migrates this config into the manifest, so that it will persist
across reboots etc.

This PR also adds Hector to the list of Github users allowed access.

## How to review

First deploy from master with OAuth enabled.
* Create an oAuth application at https://github.com/settings/applications/new
    * Authorization callback URL should be `https://deployer.<your env>.dev.cloudpipeline.digital/auth/github/callback`
* Upload the secrets to the state bucket: `GITHUB_CLIENT_ID=1234 GITHUB_CLIENT_SECRET=5678 make dev deployer-concourse upload-github-oauth`
* Push the pipelines with oauth enabled: `make dev deployer-concourse pipelines ENABLE_GITHUB=true SELF_UPDATE_PIPELINE=false`
* Run the `create-bosh-concourse` pipeline, and then verify you can login using github.

Secondly to this branch, and deploy again.
* `git checkout add_hector_and_fix_github_auth`
* Push pipelines again: `make dev deployer-concourse pipelines ENABLE_GITHUB=true SELF_UPDATE_PIPELINE=false`
* Run the `create-bosh-concourse` pipeline, and then verify you can login using github.

After the concourse instance has restarted, you should still be able to login using github auth (without having to run the rest of the pipeline).



## Who can review

Not me.
